lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -23,16 +29,41 @@ def execute
       running_in_state_machine? ? check_task_status(task_id) : wait_for_method(task_id)
     end
 
+    def process_result(task = nil)
+      stack_job(task).refresh_ems
+      log_stdout(task)
+      super
+    end
+
     def build_options_hash
       super.tap do |config_info|
         config_info[:hosts] = resolved_hosts
       end
     end
 
+    def build_launch_options
+      @build_launch_options ||= build_options_hash.slice(*CONFIG_OPTIONS_WHITELIST).tap do |options|
+        options[:hosts] = hosts_array(build_options_hash[:hosts])
+        translate_credentials!(options)
+      end
+    end
+
+    def playbook
+      @playbook ||= PLAYBOOK_CLASS.find_by(:id => @aem.options[:playbook_id])
+    end
+
+    def stack_job(task)
+      @stack_job ||= STACK_CLASS.find_by(:miq_task => task)
+    end
+
     def resolved_hosts
       @ae_object.substitute_value(@aem.options[:hosts], nil, true).tap do |value|
         raise ArgumentError, "Hosts field #{@aem.options[:hosts]} resolved to empty string" if value.blank?
       end
     end
+
+    def log_stdout(task)
+      playbook_log_stdout(@aem.options[:log_output], stack_job(task).reload)
```

Comment:
Per my other comment, I think the `reload` should be moved to the caller and not done here in a logging method.